### PR TITLE
Add invite command

### PR DESCRIPTION
**Pull request by endoffile78**
This change is [Reviewable](https://reviewable.io/reviews/utox/utox/430)

### CODE REVIEW

File: src/friend.c [hunk 1]
```diff
@@ -335,3 +335,13 @@ void friend_free(FRIEND *f)
 
     memset(f, 0, sizeof(FRIEND));//
 }
+
+FRIEND *find_friend_by_name(uint8_t *name){
+    int i;
+    for (i = 0; i < self.friend_list_count; i++) {
+        if (strcmp((const char *)friend[i].name, (const char *)name) == 0) {
```
Comment (tsudoko) [inline src/friend.c]: What if the friend has an alias set?

File: src/friend.c [hunk 2]
```diff
@@ -335,3 +335,13 @@ void friend_free(FRIEND *f)
 
     memset(f, 0, sizeof(FRIEND));//
 }
+
+FRIEND *find_friend_by_name(uint8_t *name){
+    int i;
+    for (i = 0; i < self.friend_list_count; i++) {
+        if (strcmp((const char *)friend[i].name, (const char *)name) == 0 || strcmp((const char *)friend[i].alias, (const char *)name) == 0) {
```
Comment (tsudoko) [inline src/friend.c]: Wouldn't it be better if the alias was checked first? Could prevent selecting a wrong friend in a case with multiple friends with the same names, but different aliases.

**Comment by GrayHatter** [comment]
Reviewed 1 of 4 files at r1.
Review status: 1 of 4 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/main.h, line 371 \[r3\]](https://reviewable.io:443/reviews/utox/utox/430#-KR75JD7AD8lPLGifY2H:-KR75JD8zsuDuXL-Wx7L:b-wkw4s5) ([raw file](https://github.com/utox/utox/blob/6a30148ea982be173b55dd10282297fe7d5aeffc/src/main.h#L371)):*
> ```objc
> GROUPCHAT group[MAX_NUM_GROUPS];
> 
> uint32_t friends, groups; /* Number of friends and groups */
> ```

use self.friend_list_count

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/430)*

**Comment by GrayHatter** [comment]
Reviewed 1 of 1 files at r4.
Review status: 1 of 3 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/430)*

**Comment by GrayHatter** [comment]
@endoffile78 do you want to do anything with @tsudoko's comments?

**Comment by endoffile78** [comment]
@GrayHatter I already did everything he suggested.

File: src/friend.c [hunk 3]
```diff
@@ -335,3 +335,13 @@ void friend_free(FRIEND *f)
 
     memset(f, 0, sizeof(FRIEND));//
 }
+
+FRIEND *find_friend_by_name(uint8_t *name){
+    int i;
+    for (i = 0; i < self.friend_list_count; i++) {
+        if (strcmp((const char *)friend[i].alias, (const char *)name) == 0 || strcmp((const char *)friend[i].name, (const char *)name) == 0) {
```
Comment (GrayHatter) [inline src/friend.c]: can't use string compare here, toxcore doesn't promise that names will have a null term, I know, I don't like it much either...

File: src/commands.c [hunk 4]
```diff
@@ -81,6 +81,17 @@ uint16_t utox_run_command(char_t *string, uint16_t string_length, char_t **cmd,
                 return 0;
             }
         }
+    } else if (cmd_length == 6 && memcmp(*cmd, "invite", 6) == 0) {
+        if (selected_item->item == ITEM_GROUP) {
+            GROUPCHAT *g = selected_item->data;
+            FRIEND *f = find_friend_by_name(*argument);
+            if (f != NULL && f->online) {
+                cmd_length = -1;
+                postmessage_toxcore(TOX_GROUP_SEND_INVITE, (g - group), (f - friend), NULL);
```
Comment (GrayHatter) [inline src/commands.c]: don't use `- group` or `- friend` anymore either... it's a hack that won't work when I start dynamically allocating friends...

File: src/friend.c [hunk 5]
```diff
@@ -336,10 +336,10 @@ void friend_free(FRIEND *f)
     memset(f, 0, sizeof(FRIEND));//
 }
 
-FRIEND *find_friend_by_name(uint8_t *name){
-    int i;
+FRIEND *find_friend_by_name(uint8_t *name) {
+    int i, length = strlen((char *)name);
```
Comment (GrayHatter) [inline src/friend.c]: strlen() needs the nul term too, you can use friend[i].name_length and alias_length

File: src/groups.c [hunk 6]
```diff
@@ -26,6 +26,7 @@ void group_init(GROUPCHAT *g, uint32_t group_number, _Bool av_group) {
     g->msg.panel.width          = -SCROLL_WIDTH;
     g->msg.is_groupchat         = 1;
 
+    g->number                   = group_number;
```
Comment (GrayHatter) [inline src/groups.c]: how much testing did you do on this one? I don't trust tox groups AT ALL, and I assume their main goal is to cause segfaults.

**Comment by GrayHatter** [comment]
@endoffile78 thanks mate!

**Pull request merged**